perun-core/src/main/java/cz/metacentrum/perun/taskslib/dao/jdbc/TaskResultDaoJdbc.java

```diff
@@ -287,26 +287,61 @@ public int clearOld(int engineID, int numDays) {
 	private static byte[] clearZeroBytesFromString(byte[] data, int maxLength) {
 		if (data == null) return null;
 		ByteArrayOutputStream dataOut = new ByteArrayOutputStream() ;
-		int limit = (maxLength < data.length) ? maxLength - 4 : data.length;
+		int limit = (maxLength < data.length) ? maxLength - MAX_NUMBER_OF_UTF8_BYTES : data.length;
 		for (int i = 0; i < limit; i++) {
 			if (data[i] != 0x00)
 				dataOut.write(data[i]);
 		}
 		if(maxLength < data.length) {
-			// we had to cut the byte array at limit
-			// data[limit-1] is the last added byte
-			// we have to check, if it starts the non-ASCII char sequence
-			if(data[limit-1] >= 0xC0 ) {
-				dataOut.write(data[limit]);
-			}
-			if(data[limit-1] >= 0xE0 ) {
-				dataOut.write(data[limit+1]);
-			}
-			if(data[limit-1] >= 0xF0 ) {
-				dataOut.write(data[limit+2]);
+			// check if we have to add some bytes in case we have split an UTF-8 character that is longer than 1 byte.
+			if (!isASingleByteUTF8Char(data[limit-1])) {
+				int i = 0;
+				while (!isAStartingByteUTF8Char(data[limit + i])) {
+					if (i == MAX_NUMBER_OF_UTF8_BYTES) {
+						break;
+					}
```

Comment:
Question is:
Do we want to continue when input is corrupted (not correct UTF-8) and save it anyway?

This is probably question for @zlamalp

Reply:
Add there at least "log.error" please before break. It should be enough for now. This situation should not happen.

Reply:
I have added logging of the message with appropriate information.